apps/s_client.c

```diff
@@ -2201,6 +2204,36 @@ int s_client_main(int argc, char **argv)
             BIO_read(sbio, sbuf, BUFSIZZ);
         }
         break;
+    case PROTO_SIEVE:
+        {
+            int foundit = 0;
+            BIO *fbio = BIO_new(BIO_f_buffer());
+
+            BIO_push(fbio, sbio);
+            /* wait for multi-line response to end from Sieve */
+            do {
+                mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+                /*
+                 * According to RFC 5804 § 1.7, capability
+                 * is case-insensitive, make it uppercase
+                 */
+                if (mbuf_len > 1 && mbuf[0] == '"') {
+                    make_uppercase(mbuf);
+                    if (strncmp(mbuf, "\"STARTTLS\"", 10) == 0)
+                        foundit = 1;
+                }
+            } while (mbuf_len > 1 && mbuf[0] == '"');
+            (void)BIO_flush(fbio);
+            BIO_pop(fbio);
+            BIO_free(fbio);
+            if (!foundit)
+                BIO_printf(bio_err,
+                           "Didn't find STARTTLS in server response,"
+                           " trying anyway...\n");
+            BIO_printf(sbio, "STARTTLS\r\n");
+            BIO_read(sbio, sbuf, BUFSIZZ);
```

Comment:
Question is (and one can probably ask it even in other cases) if it's worth if not paying attention to negative reply after STARTTLS, but at least print it as clue for user. I mean specification allows to reject STARTTLS request, and it's ignored here. And if request was denied and you insist on TLS handshake anyway, it will fail, but user won't be given any clue. In other words question is if it would be appropriate at least recognize negative reply and print it.